crates/accelerate/src/commutation_checker.rs

```diff
@@ -689,6 +644,26 @@ fn map_rotation<'a>(
     (op, params, false)
 }
 
+/// For a (controlled) rotation or phase gate, return a tuple ``(Tr(gate) / dim, dim)``.
+/// Returns ``None`` if the rotation gate (specified by name) is not supported.
+pub fn rotation_trace_and_dim(rotation: &str, angle: f64) -> Option<(Complex64, f64)> {
+    let dim = match rotation {
+        "rx" | "ry" | "rz" | "p" | "u1" => 2.,
+        _ => 4.,
+    };
+
+    let trace_over_dim = match rotation {
+        "rx" | "ry" | "rz" | "rxx" | "ryy" | "rzx" | "rzz" => {
+            Complex64::new((angle / 2.).cos(), 0.)
+        }
+        "crx" | "cry" | "crz" => Complex64::new(0.5 + 0.5 * (angle / 2.).cos(), 0.),
+        "p" | "u1" => (1. + Complex64::new(0., angle).exp()) / 2.,
+        "cp" => (3. + Complex64::new(0., angle).exp()) / 4.,
+        _ => return None,
```

Comment:
I feel like we could do this without the strings and used the `StandardGate` enum variants from standard gates. It would be faster to do the enum match because it's a u8 vs a string comparison.

Reply:
I just saw this is used elsewhere in accelerate. I feel like this comment is especially true if we're going to use this for standardized trace computation for standard gates.